infra/aws/terraform/registry.k8s.io/s3/bucket.tf

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_s3_bucket" "registry-k8s-io" {
+  bucket = "${var.prefix}registry-k8s-io-${var.region}"
+}
+
+resource "aws_s3_bucket_acl" "registry-k8s-io" {
+  bucket = aws_s3_bucket.registry-k8s-io.bucket
+  acl    = "public-read"
+}
+
+resource "aws_s3_bucket_ownership_controls" "registry-k8s-io" {
+  bucket = aws_s3_bucket.registry-k8s-io.bucket
+
+  rule {
+    object_ownership = "BucketOwnerEnforced"
+  }
+}
+
+resource "aws_s3_bucket_policy" "registry-k8s-io-public-read" {
+  bucket = aws_s3_bucket.registry-k8s-io.bucket
+
+  policy = jsonencode({
+    "Id" : "Public-Access",
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : "s3:GetObject",
+        "Effect" : "Allow",
+        "Resource" : "${aws_s3_bucket.registry-k8s-io.arn}/*",
+        "Principal" : "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_user_policy" "registry-k8s-io-rw" {
+  name = "${var.prefix}${aws_s3_bucket.registry-k8s-io.bucket}-access"
+  user = var.iam_user_name
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:ListBucket",
+          "s3:DeleteObject"
```

Comment:
Since the `Resource` here refers to just the bucket resource, I believe the only permission you need in this block is:

`s3:ListBucket`

since all of the other permissions are only applicable to the `Object` resources referenced by the `Resource` string `"${aws_s3_bucket.registry-k8s-io.arn}/*"`

Reply:
The intent of this _aws_iam_user_policy_ is to allow the IAM user created in the above folder's _accounts.tf_ to r/w the bucket for CI related things. The IAM user would be the one that is used with the promotion tools.

Should be permissions still be reduced here?

Reply:
@BobyMCbobs The IAM user that is writing objects to these buckets needs these permissions:

On the bucket resource (i.e. `"${aws_s3_bucket.registry-k8s-io.arn}/"`:

* `s3:ListBucket`

On the object resources *within* that bucket (i.e. `"${aws_s3_bucket.registry-k8s-io.arn}/*"`:

* `s3:PutObject`
* `s3:GetObject`
* `s3:DeleteObject`

Hope that makes sense! :)

Reply:
Thank you @jaypipes, I've updated the permissions in 3a99bd522